var/spack/repos/builtin/packages/abseil-cpp/package.py

```diff
@@ -27,8 +27,15 @@ class AbseilCpp(CMakePackage):
 
     conflicts('+shared', when='@:20190808')
 
+    variant('cxxstd', values=('11', '14', '17', '20'), default='11',
+            description="C++ standard used during compilation")
+
     def cmake_args(self):
-        args = ["-DBUILD_TESTING=OFF",  "-DCMAKE_CXX_STANDARD=11"]
+        cxxstd = self.spec.variants['cxxstd'].value
+        args = [
+            "-DBUILD_TESTING=OFF",
+            "-DCMAKE_CXX_STANDARD={}".format(cxxstd)
+        ]
```

Comment:
I think you can rewrite this entire part as:
```python
    return [
        self.define('BUILD_TESTING', 'OFF'),
        self.define('CMAKE_CXX_STANDARD', cxxstd),
        self.define_from_variant('BUILD_SHARED_LIBS', 'shared')
    ]
```
and remove the `args.append` below.